ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp

```diff
@@ -183,18 +174,49 @@ std::unique_ptr<Process> createHydroMechanicsProcess(
     }
 
     // Reference temperature
-    const auto& reference_temperature =
+    double const reference_temperature =
         //! \ogs_file_param{prj__processes__process__HYDRO_MECHANICS__reference_temperature}
-        config.getConfigParameter<double>(
-            "reference_temperature", std::numeric_limits<double>::quiet_NaN());
+        config.getConfigParameter<double>("reference_temperature", 1);
+    DBUG("Use '%s' as reference temperature.", "reference_temperature");
+
+    //Specific gas constant
+    double const specific_gas_constant =
+        //! \ogs_file_param{prj__processes__process__HYDRO_MECHANICS__specific_gas_constant}
+        config.getConfigParameter<double>("specific_gas_constant", 1);
+    DBUG("Use '%s' as specific gas constant.", "specific gas constant");
+
+    // Fluid compressibility
+    double const fluid_compressibility =
+        //! \ogs_file_param_special{prj__processes__process__HYDRO_MECHANICS__fluid_compressibility}
+        config.getConfigParameter<double>("fluid_compressibility", 0);
+    DBUG("Use '%s' as fluid compressibility parameter.", "fluid_compressibility");
+
+    auto const& fluid_type_str =
+        //! \ogs_file_param{prj__processes__process__HYDRO_MECHANICS__fluid_type}
+        config.getConfigParameter<std::string>("fluid_type", "incompressible_fluid");
+
+    unsigned const fluid_type =
+        (fluid_type_str == "incompressible_fluid") ? 0 :
+        (fluid_type_str == "compressible_fluid") ? 1 :
+        (fluid_type_str == "ideal_gas") ? 2 : 3;
+
+    if (fluid_type == 3)
+    {
+        OGS_FATAL(
+            "The fluid type %s is unavailable.\n"
+            "The available fluid types are ideal_gas and "
```

Comment:
Above also `compressible_fluid` type is handled, but not mentioned in the error message.